### PR TITLE
Undo/Redo v1.2

### PR DESCRIPTION
**Pull request by brebeek**
Implemented undo/redo feature to Typee.
Previous states of the list are stored as a list in HistoryManager, and the current version is updated by a pointer.
Please help in reviewing thanks!

### CODE REVIEW

File: src/main/java/com/typee/model/ModelManager.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs
         this.addressBook = new AddressBook(addressBook);
         this.statedAppointmentList = new StatedAddressBook(addressBook);
         this.userPrefs = new UserPrefs(userPrefs);
-        filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
+        filteredPersons = new FilteredList<>(this.statedAppointmentList.getPersonList());
```
Comment (nordic96) [inline src/main/java/com/typee/model/ModelManager.java]: perhaps can refactor the attribute filteredPersons to filteredAppointments or filteredEngagements. To reduce confusions in the future.

File: src/main/java/com/typee/model/ModelManager.java [hunk 2]
```diff
@@ -85,36 +85,36 @@ public void setAddressBookFilePath(Path addressBookFilePath) {
 
     @Override
     public void setAddressBook(ReadOnlyAddressBook addressBook) {
-        this.addressBook.resetData(addressBook);
+        this.statedAppointmentList.resetData(addressBook);
```
Comment (nordic96) [inline src/main/java/com/typee/model/ModelManager.java]: in our next update, we should rename all adressbook related attributes and methods to typee. But as of now, i think it is fine to keep it this way as long it is working. good job.

Comment (brebeek) [reply src/main/java/com/typee/model/ModelManager.java]: Agreed. I'll work on refactoring at a later time.

**Comment by jun-ha0** [comment]
LTGM

**Comment by brebeek** [comment]
resolves #18 #25 #28

**Pull request merged**